sdg/meta.py

```diff
@@ -11,12 +11,20 @@
 import sdg
 from sdg.path import input_path, output_path  # local package
 
-def read_meta(inid, git=True, src_dir='', git_data_dir=None):
+def read_meta(inid, path_pattern='', git=True, src_dir='', git_data_dir=None):
     """Perform pre-processing for the metadata files"""
     status = True
     # Read and write paths may be different
-    fr = input_path(inid, ftype='meta', src_dir=src_dir)
-
+    if path_pattern == '':
+        meta_folder = 'meta'
+        extension='.md'
+    else:
+        meta_folder = path_pattern.split("/")[0]
```

Comment:
After looking a bit more, I think around here is where the problematic bits might be. A couple of issues maybe:
1. Instead of hardcoding "/" we should use `os.path.sep` (easy fix)
2. Bigger issue - this won't work if the path_pattern has multiple parts, like `path/to/my/meta/*-*.md` for example. I think maybe this is related to why the tests are failing. ([Here is the path pattern for the test suite.](https://github.com/open-sdg/sdg-build/blob/master/tests/test_build.py#L26))

Reply:
@brockfanning thanks for taking a look. What id we use os.path.split e.g. see here:
https://code.sololearn.com/crI157thO5WX/#py

Fixes hardcoding and takes all parts except for last part i.e. gets the folder paths without the file name

Reply:
@LucyGwilliamAdmin That sounds good to me.